runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -12358,6 +12358,936 @@ static TR::Register *inlineIntrinsicIndexOf(TR::Node *node, TR::CodeGenerator *c
+    * |                1 |           0 |           0 |           0 |            0 | The one character in the substring is a known match so no further loads are needed.
+    * |                2 |           1 |             |             |              |
+    * |                3 |           1 |             |             |              | Recomparing the first byte isn't needed so a single 2 byte load handles the rest of the substring.
+    * |                4 |             |           1 |             |              |
+    * |                5 |             |           1 |             |              |
+    * |                6 |           1 |           1 |             |              |
+    * |                7 |           1 |           1 |             |              |
+    * |                8 |             |             |           1 |              |
+    * |                9 |             |             |           1 |              |
+    * |               10 |           1 |             |           1 |              |
+    * |               11 |           1 |             |           1 |              |
+    * |               12 |             |           1 |           1 |              |
+    * |               13 |             |           1 |           1 |              |
+    * |               14 |           1 |           1 |           1 |              |
+    * |               15 |           1 |           1 |           1 |              |
+    * |               16 |             |             |             |            1 |
+    * |               17 |             |             |             |            1 |
+    */
+   TR::LabelSymbol *load16LoopLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *skipLoad16Label = generateLabelSymbol(cg);
+   TR::LabelSymbol *skipLoad8Label = generateLabelSymbol(cg);
+   TR::LabelSymbol *skipLoad4Label = generateLabelSymbol(cg);
+
+   /*
+    * subStringMismatchLabel is the location to jump to if the substring does not match the next characters in the main string.
+    * This label can be provided by the caller of performArrayCmp if there is a specific location to jump to.
+    * If it isn't, createSubStringMismatchLabel is generated at the end of the assembly made by performArrayCmp.
+    */
+   bool createSubStringMismatchLabel = false;
+   if (nullptr == subStringMismatchLabel)
+      {
+      subStringMismatchLabel = generateLabelSymbol(cg);
+      createSubStringMismatchLabel = true;
+      }
+
+   /*
+    * If subStringLenReg is even, tempReg is equal to the length of the substring.
+    * If subStringLenReg is odd,  tempReg is now 1 less than the length of the substring.
+    * This is done so one fewer byte is compared in odd cases.
+    * tempReg holds the actual number of bytes that need to be compared and is always an even number.
+    * If tempReg is less than 2, it must be 0 so a full substring match has been found.
+    */
```

Comment:
observation of inconsistency:  offset, length, and index values seemed being treated as 64bit in this area of code, but not in others. preference: treating them consistently.

Reply:
I can try looking over this.